docs/proposals/migration/design-of-seamless-cluster-migration-scheme.md

```diff
@@ -0,0 +1,316 @@
+
+
+> Detailed practical steps can be found in: [Utilizing existing policy capabilities to achieve cluster migration](https://www.yuque.com/chaosi/pwzt9c/ypugrp6452lu36ok)
+
+
+
+
+The above methods all require **the migration of specific native resources that the operator perceives in the member cluster**. In some `CI/CD` automated pipeline scenarios, **the operator cannot manually modify native resource declarations such as `Deployment`, and the resources contained in the original cluster are also not visible to the operator**. This brings difficulties and challenges to achieving seamless migration.
+
+
+
+### Goals
+
+Explore how to enable `Karmada`  to achieve seamless cluster migration in a more perfect and universal way, to meet more complex user scenarios and achieve complete user insensitivity?
+
+In detail, we are proposed to extend the API semantics of Karmada custom resources to support declaring how to take over a matching resource when there is a conflict in the member cluster.
+
+
+
+### Non-Goals
+
+If there is a conflict in the member cluster and the resource is not taken over during the conflict, how to ensure that the total number of replicas meets the expected `ResourceTemplate`? This follows the original logic and is not within the scope of this proposal discussion.
+
+> For example, the `ResourceTemplate` specifies that two replicas need to be evenly allocated to two member clusters. If member cluster 1 fails to allocate due to conflicts and does not take over, the proposal does not care about how to handle the failure issue.
+
+
+
+## Proposal
+
+1. For custom resources such as `PropagationPolicy`, `ResourceBinding`, and `Work`, add a field called `conflictResolution`. The operator can use this field in the `PropagationPolicy` to specify how to take over the matched resource when there is a conflict in the member cluster.
+
+2. The field `conflictResolution` has the following values:
+
+   * `overwrite` : continue to propogate the resource and overwrite the new configuration to member clusters
+
+   * `skip-propagate`: skip the propagating process and retain the previous configuration in member clusters (default value)
+
+     
+
+> Points worth supplementing：for native resources such as `ResourceTemplate`, preserve the `work.karmada.io/conflict-resolution: overwrite` annotation method. Users can add `conflict-resolution` annotations for specific resources in the `ResourceTemplate` to override the `conflictResolution` configuration of the `PropagationPolicy`.
+
+
+
+### User Stories (Optional)
+
+#### Story 1
+
+Assuming that there are many deployments in the original single cluster, and the user wants `Karmada` to fully take over all deployments.
```

Comment:
I think it's better to point out what's the situation now and what users expect. Give an intuitive comparison.

Reply:
updated